internal/pubsub/pubsub.go

```diff
@@ -197,7 +179,11 @@ func (s *Subscription) getNextBatch(ctx context.Context) error {
 			Body:     m.Body,
 			Metadata: m.Metadata,
 			ack: func() {
-				s.ackBatcher.Add(ackIDBox{id}, size)
+				s.wg.Add(1)
+				go func() {
+					defer s.wg.Done()
+					s.ackBatcher.Add(ctx, ackIDBox{id})
```

Comment:
Change to AddNoWait after #807. You'll still need to deal with the errors somehow.

Reply:
Done. I don't know what to do with the errors. With blocking acks it was straightforward. Now I have no idea.